04.class/memo_app.js

```diff
@@ -0,0 +1,121 @@
+import * as fs from "node:fs";
+import Enquirer from "enquirer";
+import { v4 as uuidv4 } from "uuid";
+
+const path = "memos.json";
+
+export class MemosApp {
+  constructor() {
+    if (!fs.existsSync(path)) {
+      this.createFile();
+    }
+    this.baseData = JSON.parse(fs.readFileSync(path, "utf-8"));
+  }
+
+  createFile() {
+    const rawData = {};
+    const json = JSON.stringify(rawData);
+
+    fs.writeFileSync(path, json, function (err) {
+      if (err) {
+        throw err;
+      }
+    });
+  }
+
+  add() {
+    const uuid = uuidv4();
+    const input = fs.readFileSync("/dev/stdin", "utf-8");
+    const memo = input.split("\n");
+    memo.pop();
+
+    if (memo[0] === "") {
+      console.log("メモの1行目を入力してください。");
+      return;
+    }
+
+    this.baseData[uuid] = memo;
+
+    const json = JSON.stringify(this.baseData);
+    fs.writeFile(path, json, function (err) {
+      if (err) {
+        throw err;
+      }
+    });
+    console.log("メモを保存しました。");
+  }
+
+  list() {
+    for (const property in this.baseData) {
```

Comment:
```suggestion
    for (const uuid in this.baseData) {
```

とした方がわかりやすそうです（ここ以外も同様）

Reply:
こちらも考慮が足りていませんでした...
変数名を修正いたしました！